### DISCUSSION TITLE
Adding "AdditionalReferencedDocument" within SpecifiedLineTradeSettlement.

### DISCUSSION THREAD

**Discussion by PendragonCS** [stephanstapel/ZUGFeRD-csharp#229] [Q&A]
Hi All,

I've been playing around with this solution with the aim to use it to generate xrechnung_2.3 files.

But i can't seem to get the AdditionalReferencedDocument elements within the SpecifiedLineTradeSettlement node which is part of this standard.

I've tried adding them direct and also using the call 'AddAdditionalReferencedDocument' found on the created tradelineitem.  They do appear to be added to the object but do not form part of the generated XML.

Is there a specific way of doing this?

Thanks

Jason

**Comment by stephanstapel** [comment]
The library so far supports additional reference documents on line item level for SpecifiedLineTradeAgreement only. This is where the AdditionalReferenceDocuments you created should appear in your sample file.

Honestly speaking, I have no idea how to implement the possibility to attach additional document references at such many places in the document and still leave the library usable to the average user.

**Reply by PendragonCS** [reply]
Thanks Stephan 

I can see in the code holder for this function but appreciate that its may be difficult to fully implement that.

Regards

**Reply by stephanstapel** [reply]
Yep, at least the comment is there already :)
The xml structure itself is simple, the code is already there at other places. What I would like to avoid with the library is to replicate the xml structure with c# classes so I don't have an idea how to add a second list of additionalreferencedocuments to the line item class
